packages/client/src/schedule-types.ts

```diff
@@ -0,0 +1,845 @@
+ * interpreted as equal to start. Similarly, if step is less than 1, then step is interpreted as 1.
+ *
+ * @experimental
+ */
+export interface Range<Unit> {
+  /**
+   * Start of range (inclusive)
+   */
+  start: Unit;
+
+  /**
+   * End of range (inclusive)
+   *
+   * @default `start`
+   */
+  end: Unit;
+
+  /**
+   * The step to take between each value.
+   *
+   * @default 1
+   */
+  step: number;
+}
+
+/**
+ * A {@link Range} definition, with support for loose syntax.
+ *
+ * For example:
+ * ```
+ * 3 ➡️ 3
+ * { start: 2 } ➡️ 2
+ * { start: 2, end: 4 } ➡️ 2, 3, 4
+ * { start: 2, end: 10, step: 3 } ➡️ 2, 5, 8
+ * ```
+ *
+ * @experimental
+ */
+export type LooseRange<Unit> =
+  | Range<Unit>
+  | { start: Range<Unit>['start']; end?: Range<Unit>['end']; step?: never }
+  | Unit;
+
+/** @experimental */
+export const Months = [
+  'JANUARY',
+  'FEBRUARY',
+  'MARCH',
+  'APRIL',
+  'MAY',
+  'JUNE',
+  'JULY',
+  'AUGUST',
+  'SEPTEMBER',
+  'OCTOBER',
+  'NOVEMBER',
+  'DECEMBER',
+] as const;
+
+/** @experimental */
+export type Month = typeof Months[number];
+
+/** @experimental */
+export const DaysOfWeek = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'] as const;
+
+/** @experimental */
+export type DayOfWeek = typeof DaysOfWeek[number];
+
+/** @experimental */
+export type ScheduleOptionsAction = ScheduleOptionsStartWorkflowAction<Workflow>;
+
+/** @experimental */
+export type ScheduleOptionsStartWorkflowAction<W extends Workflow> = {
+  type: 'startWorkflow';
+  workflowType: string | W;
+} & Pick<
+  WorkflowStartOptions<W>,
+  | 'taskQueue'
+  | 'workflowId'
+  | 'args'
+  | 'memo'
+  | 'searchAttributes'
+  | 'retry'
+  | 'workflowExecutionTimeout'
+  | 'workflowRunTimeout'
+  | 'workflowTaskTimeout'
+>;
+
+/** @experimental */
+export type ScheduleSummaryAction = ScheduleSummaryStartWorkflowAction;
+
+/** @experimental */
+export interface ScheduleSummaryStartWorkflowAction {
+  type: 'startWorkflow';
+  workflowType: string;
+}
+/** @experimental */
+export type ScheduleDescriptionAction = ScheduleDescriptionStartWorkflowAction;
+
+/** @experimental */
+export type ScheduleDescriptionStartWorkflowAction = ScheduleSummaryStartWorkflowAction &
+  Pick<
+    WorkflowStartOptions<Workflow>,
+    | 'taskQueue'
+    | 'workflowId'
+    | 'args'
+    | 'memo'
+    | 'searchAttributes'
+    | 'retry'
+    | 'workflowExecutionTimeout'
+    | 'workflowRunTimeout'
+    | 'workflowTaskTimeout'
+  >;
+
+// Invariant: an existing ScheduleDescriptionAction can be used as is to create or update a schedule
+checkExtends<ScheduleOptionsAction, ScheduleDescriptionAction>();
+
+/** @experimental */
+export type CompiledScheduleAction = Replace<
+  ScheduleDescriptionAction,
+  {
+    workflowType: string;
+    args: unknown[];
+  }
+>;
+
+/**
+ * Policy for overlapping Actions.
+ *
+ * @experimental
+ */
+export enum ScheduleOverlapPolicy {
```

Comment:
Still not liking the inconsistency here with the rest of the SDK.
I'm not blocking on this but it deserves a short discussion IMHO.

Reply:
Here's the proposal discussion: https://github.com/temporalio/proposals/pull/62#discussion_r942844591

Reply:
The only thing that bothers me with reexporting the proto's enum directly is that it doesn't provide type doc. The values in this enum are not trivial to understand for users, so that doc is particularly important.